autometa/common/external/diamond.py

```diff
@@ -343,78 +518,104 @@ def add_taxids(hits, database, verbose=True):
+        )
     n_lines = file_length(database) if verbose else None
-    desc = f'Parsing {os.path.basename(database)}'
+    desc = f"Parsing {os.path.basename(database)}"
     acc2taxids = {}
     for line in tqdm(fh, disable=disable, desc=desc, total=n_lines, leave=False):
-        line = line.decode() if is_gzipped else line
-        acc_num, acc_ver, taxid, _ = line.split('\t')
+        acc_num, acc_ver, taxid, _ = line.split("\t")
         taxid = int(taxid)
         if acc_num in accessions:
-            acc2taxids.update({acc_num:taxid})
+            acc2taxids.update({acc_num: taxid})
         if acc_ver in accessions:
-            acc2taxids.update({acc_ver:taxid})
+            acc2taxids.update({acc_ver: taxid})
     fh.close()
     n_qseqids = len(hits)
-    desc = f'Translating {n_qseqids} qseqids\' accessions to taxids'
-    for qseqid,hit in tqdm(hits.items(), disable=disable, total=n_qseqids, desc=desc, leave=False):
+    desc = f"Translating {n_qseqids} qseqids' accessions to taxids"
+    ncbi = NCBI(NCBI_DIR, verbose)  # Instantiating the NCBI class
+    for qseqid, hit in tqdm(
+        hits.items(), disable=disable, total=n_qseqids, desc=desc, leave=False
+    ):
         for sseqid in hit.sseqids:
             taxid = acc2taxids.get(sseqid)
-            # if taxid is None:
-            #     raise DatabasesOutOfDateError(f'{sseqid} deprecated/suppressed/removed')
-            hit.sseqids[sseqid].update({'taxid':taxid})
+            if not taxid:
+                taxid = ncbi.merged.get(taxid, taxid)
+            hit.sseqids[sseqid].update({"taxid": taxid})
     return hits
 
-def main(args):
+
+def main():
+    import argparse
+    import os
+
+    parser = argparse.ArgumentParser(
+        description="""
+    Retrieves blastp hits with provided input assembly
+    """
+    )
+    parser.add_argument("fasta", help="</path/to/faa/file>")
+    parser.add_argument("database", help="</path/to/diamond/formatted/database>")
+    parser.add_argument("acc2taxids", help="</path/to/ncbi/prot.accession2taxid.gz>")
+    parser.add_argument("outfile", help="</path/to/diamond/output/file>")
+    parser.add_argument(
+        "blast_type",
+        help="[blastp]: A.A -> A.A. [blastx]: Nucl. -> A.A.",
+        default="blastp",
+        choices=["blastp", "blastx"],
+    )
+    parser.add_argument(
+        "--evalue", help="diamond evalue threshold", default=float("1e-5")
+    )
+    parser.add_argument(
+        "--maxtargetseqs",
+        help="max target sequences to retrieve per query",
+        default=200,
+        type=int,
+    )
+    parser.add_argument(
+        "--nproc", help="number of processors to use", default=mp.cpu_count(), type=int
+    )
+    parser.add_argument("--tmpdir", help="</path/to/tmp/directory>", default=os.curdir)
+    parser.add_argument(
+        "--bitscr-thresh",
+        help="threshold above which bitscores will be selected",
+        default=0.9,
+    )
```

Comment:
This is an end-user argument and should be more descriptive. e.g. `--bitscore-threshold`. The help text is not very informative